components/script/dom/document.rs

```diff
@@ -3853,18 +3844,28 @@ impl DocumentMethods for Document {
 
     // https://html.spec.whatwg.org/multipage/#dom-document-getelementsbyname
     fn GetElementsByName(&self, name: DOMString) -> DomRoot<NodeList> {
-        self.create_node_list(|node| {
-            let element = match node.downcast::<Element>() {
-                Some(element) => element,
-                None => return false,
-            };
-            if element.namespace() != &ns!(html) {
-                return false;
-            }
-            element
-                .get_attribute(&ns!(), &local_name!("name"))
-                .map_or(false, |attr| &**attr.value() == &*name)
-        })
+        let doc = self.GetDocumentElement();
```

Comment:
Why using the document element instead of `self`? A document is a node as well.

Reply:
Yeah, I think there wasn't any reason why the document element needs to be used. Let's update this.